include/range/v3/view/indices.hpp

```diff
@@ -0,0 +1,117 @@
+                (
+                    detail::take_exactly_view_<iota_view<Val>, true>
+                        {iota_view<Val>{from}, detail::ints_distance_(from, to)}
+                )
+
+                template<typename Val, CONCEPT_REQUIRES_(Integral<Val>())>
+                auto operator()(Val to) const
+                RANGES_DECLTYPE_AUTO_RETURN_NOEXCEPT
+                (
+                    (*this)(Val(), to)
+                )
+
+            #ifndef RANGES_DOXYGEN_INVOKED
+                template<typename Val, CONCEPT_REQUIRES_(!Integral<Val>())>
+                void operator()(Val) const
+                {
+                    CONCEPT_ASSERT_MSG(Integral<Val>(),
+                        "The object passed to view::indices must be Integral");
+                }
+                template<typename Val, CONCEPT_REQUIRES_(!Integral<Val>())>
+                void operator()(Val, Val) const
+                {
+                    CONCEPT_ASSERT_MSG(Integral<Val>(),
+                        "The object passed to view::indices must be Integral");
+                }
+            #endif
+            };
+
+            /// Inclusive range of indices: [from, to].
+            struct inclusive_indices_fn
+              : iota_view<std::ptrdiff_t>
+            {
+                template<typename Val, CONCEPT_REQUIRES_(Integral<Val>())>
+                auto operator()(Val from, Val to) const
+                RANGES_DECLTYPE_AUTO_RETURN_NOEXCEPT
+                (
+                    detail::take_exactly_view_<iota_view<Val>, true>
+                        {iota_view<Val>{from}, detail::ints_distance_(from, to) + 1}
+                )
+
+                template<typename Val, CONCEPT_REQUIRES_(Integral<Val>())>
+                auto operator()(Val to) const
+                RANGES_DECLTYPE_AUTO_RETURN_NOEXCEPT
+                (
+                    (*this)(Val(), to)
+                )
+
+            #ifndef RANGES_DOXYGEN_INVOKED
+                template<typename Val, CONCEPT_REQUIRES_(!Integral<Val>())>
+                void operator()(Val) const
+                {
+                    CONCEPT_ASSERT_MSG(Integral<Val>(),
+                        "The object passed to view::inclusive_indices must be Integral");
+                }
+                template<typename Val, CONCEPT_REQUIRES_(!Integral<Val>())>
+                void operator()(Val, Val) const
+                {
+                    CONCEPT_ASSERT_MSG(Integral<Val>(),
+                        "The object passed to view::inclusive_indices must be Integral");
+                }
+            #endif
+            };
+
+            /// \relates indices_fn
+            /// \ingroup group-views
+            RANGES_INLINE_VARIABLE(indices_fn, indices)
+
+            /// \relates inclusive_indices_fn
+            /// \ingroup group-views
+            RANGES_INLINE_VARIABLE(inclusive_indices_fn, inclusive_indices)
+
+        }  // namespace view
+    }
+}
+
```

Comment:
IIUC `indices` is not a "view", it just returns `iota | take_exactly` and `iota` already does this, so that should be enough right?